### ISSUE TITLE
Question: rate limits/obtaining all DOIs from a search

### ISSUE THREAD

**Issue by banderson10** [ropensci/fulltext#240]
Hello,

I am using the ft_search() function within the fulltext package. I am specifically searching the Wiley database (which recently replaced the Crossref TDM database). I have obtained my API. I understand there are rate limits on the Wiley database. Using the search shown below, I observe that 70,244 articles are found using ft_search(). Due to the rate limit imposed by Wiley, I can only obtain 1000 DOIs at a time. 

opts1 <- list(key="xxx")
search1 <- ft_search(query = 'spanish flu', from = 'crossref', wileyopts=opts1, limit=1000)

In the user manual for fulltext (p. 4), it states that Wiley will only return 3 requests per second. So, after I wait 5-10 minutes, I run another search as shown below:

search2 <- ft_search(query = 'spanish flu', from = 'crossref', wileyopts=opts1, limit=1000)

I create data frames that contain the DOIs from both search1 and search2. I notice that the DOIs are the same. I have two questions:

1. How do you obtain different/unique DOIs from the first and second searches?

2. Is there a more efficient way of using ft_search(), given that the limit is 1000 for each search, and I need to obtain 70,244 articles. Is there a way I can incorporate a timing element to ft_search(), for example, every 3 minutes, go pull another 10000 DOIs?

Thank you for any advice you can provide.

Billie

**Comment by sckott** [comment]
Please share the output of `sessionInfo()` after loading `fulltext` so I can better help you

> which recently replaced the Crossref TDM database

what do you mean? I'm not aware of this

Which API key are you talking about? 

`ft_search` DOES NOT have an option to search Wiley. With respect to Wiley, you may be thinking of [`ft_get`](https://docs.ropensci.org/fulltext/reference/ft_get.html) which does integrate with Wiley

If your API key is for the Crossref Text and Data Mining thing, that program is being shut down. 

See here https://docs.ropensci.org/fulltext/reference/ft_search.html#pagination for details on pagination options with `ft_search` - for Crossref you'd want to use `offset` to start each subsequent search after the first one at a larger number. e.g, 

```r
# first request
opts = list(limit=1000, offset=0)
ft_search(query = 'spanish flu', from = 'crossref', crossrefopts = opts)

# second request
opts = list(limit=1000, offset=1000)
ft_search(query = 'spanish flu', from = 'crossref', crossrefopts = opts)

# and so on
```

**Comment by banderson10** [comment]
Thank you for your reply, Scott. This was my mistake, and I did not understand what I had misunderstood until you responded. I was confusing the information regarding the Wiley TDM key with how to use Crossref in the ft_search() function. Your comments regarding pagination makes perfect sense, and I have read the link you sent me regarding pagination. With this link along with your code snippet, I have been able to develop the code I need to access the articles in Crossref.

Thank you very much!

Billie
